garage/tf/regressors/base2.py

```diff
@@ -0,0 +1,175 @@
+"""Regressor base classes without Parameterized."""
+import abc
+
+import tensorflow as tf
+
+from garage.misc.tensor_utils import flatten_tensors
+from garage.misc.tensor_utils import unflatten_tensors
+
+
+class Regressor2(abc.ABC):
+    """
+    Regressor base class without Parameterized.
+
+    Args:
+        input_shape (tuple): Input shape.
+        output_dim (int): Output dimension.
+        name (str): Name of the regressor.
+    """
+
+    def __init__(self, input_shape, output_dim, name):
+        self._input_shape = input_shape
+        self._output_dim = output_dim
+        self._name = name
+        self._variable_scope = None
+        self._cached_params = {}
+        self._cached_param_dtypes = {}
+        self._cached_param_shapes = {}
+        self._cached_assign_ops = {}
+        self._cached_assign_placeholders = {}
+
+    def fit(self, xs, ys):
+        """
+        Fit with input data xs and label ys.
+
+        Args:
+            xs: Input data.
+            ys: Label of input data.
+        """
+        raise NotImplementedError
+
+    def predict(self, xs):
+        """
+        Predict ys based on input xs.
+
+        Args:
+            xs: Input data.
```

Comment:
Return value?

Reply:
types, please

Reply:
Usually the input and output for this kind of general API would be numpy.ndarray?

Reply:
i think so. if there's more information on the appropriate shape, please include it in the docstring.

Reply:
Fixed. If everything looks good, I will squash and merge.

Reply:
lgtm